euphonic_wrapper.py

```diff
@@ -0,0 +1,194 @@
+import numpy as np
+import os
+from euphonic import Crystal, ForceConstants, QpointPhononModes, DebyeWaller, ureg
+from euphonic.util import mp_grid
+
+class EuphonicWrapper(object):
+    # This a wrapper around the Euphonic ForceConstants and QpointPhononModes classes to make it easier to access from Matlab
+    # It is meant to be used with a Matlab python_wrapper class and implements a horace_sqw function for use with Horace
+
+    defaults = {'force_constants': None, 'phonon_modes': None, 'debye_waller': None, 'debye_waller_grid': None, 
+                'temperature': 0.0 * ureg('K'), 'bose': True, 'negative_e': False, 'conversion_mat': None, 'chunk': 5000, 
+                'lim': np.inf, 'scattering_lengths': 'Sears1992', 'weights': None, 'asr': None, 'dipole': True, 
+                'eta_scale': 1.0, 'splitting': True, 'insert_gamma': False, 'reduce_qpts': True, 'use_c': False, 
+                'n_threads': 1, 'fall_back_on_python': True, 'convolution_function': 'gauss'}
+
+    def __init__(self, value=None, **kwargs):
+        for key, val in self.defaults.items():
+            setattr(self, key, kwargs.pop(key, self.defaults[key]))
+
+        if value is not None:
+            if hasattr(value, 'calculate_qpoint_phonon_modes'):
+                self.force_constants = value
+            elif hasattr(value, 'calculate_structure_factor'):
+                self.phonon_modes = value
+            elif isinstance(value, dict):
+                kwargs.update(value)
+
+    def calculate_sf(self, qpts):
```

Comment:
Sure, I'll make it private.

I guess I just wanted users to have as much freedom as possible, but I guess the plethora of methods would be confusing to people. 

In any case I've overloaded the Matlab wrapper class so that you can access private methods with braces - e.g.

```matlab
fc = euphonic.force_constants.from_castep('quartz/quartz.castep_bin');
fc{'_calculate_gamma_correction'}([1;2;3])
```

Of course the tab completion doesn't work for this so users have no idea what the private methods names are - they'd have to look in the source (or use `py.dir(fc.pyobj)` I guess...) so it's probably ok - I wanted to give users as much freedom as possible... (so hiding it like this should make it possible for advance users to access but keep it away from beginners).

Reply:
I think that's probably best, if there's only one public method then its obvious which one to use! It's easier to start off with everything private and make useful things public rather than the other way around...